lib/query-interface.js

```diff
@@ -983,17 +983,26 @@ class QueryInterface {
    *    updatedAt: new Date()
    *  }]);
    *
+   * @param {Model}  model       Model class
    * @param {string} tableName   Table name to insert record to
    * @param {Array}  records     List of records to insert
    * @param {Object} options     Various options, please see Model.bulkCreate options
    * @param {Object} attributes  Various attributes mapped by field name
    *
    * @returns {Promise}
    */
-  bulkInsert(tableName, records, options, attributes) {
+  bulkInsert(model, tableName, records, options, attributes) {
```

Comment:
This changes the signature of this method which is part of public api. Caller should pass `options.upsertKeys` directly, add this new option `options.upsertKeys` to api docs and typings

Reply:
I think I figured out a way to do this without changing the method signature in commit 8381105. I moved the logic that I needed the `model` for in to `Model.bulkCreate()` instead of `bulkInsert()`.